policy-controller/src/admission.rs

```diff
@@ -384,3 +389,69 @@ impl Validate<ServerAuthorizationSpec> for Admission {
         Ok(())
     }
 }
+
+#[async_trait::async_trait]
+impl Validate<HttpRouteSpec> for Admission {
+    async fn validate(self, _ns: &str, _name: &str, spec: HttpRouteSpec) -> Result<()> {
+        if let Some(parent_refs) = spec.inner.parent_refs {
+            // Only validate HttpRoutes which have a Server as a parent_ref.
+            if parent_refs.iter().any(|parent_ref| {
```

Comment:
The use of `any` here means that it will stop as soon as it finds its first `parent_ref` that is a Server.

We should probably make sure that every `parent_ref` Server is valid—that is if it's even okay for multiple Servers to be referenced. If not, it should at least error if there are multiple Servers.

Reply:
We're not really validating the parent_refs here, it's more that we only want to validate the HttpRoute as a whole if it's attached to a Server.  Thus, we go forward with validation if at least one parent_ref is a Server, and therefore I think using `any` is right.  An HttpRoute can be attached to multiple Servers (or even different parents, potentially) so having multiple Server parent_refs should be fine.

Reply:
Oh right yea we're validating `spec.rules` here which doesn't have to do with `parent_ref`—makes sense then.